konstraints-core/src/main/kotlin/tools/aqua/konstraints/theories/ArraysEx.kt

```diff
@@ -25,6 +25,8 @@ import tools.aqua.konstraints.smt.*
 class ArraySort<X : Sort, Y : Sort>(val x: X, val y: Y) :
     Sort("Array".symbol(), emptyList(), listOf(x, y)) {
   override fun toString(): String = "(Array $x $y)"
+
+  override val theories = setOf(Theories.ARRAYS_EX)
```

Comment:
This construction results in hundreds of single-element sets being constructed -- can we improve that?

what about an interface that handles that; this would remove tons of duplicates and could cache the set